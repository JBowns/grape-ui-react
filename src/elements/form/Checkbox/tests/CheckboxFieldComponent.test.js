import React from 'react';
import { CheckboxFieldComponent } from '../';
import 'jest-styled-components';
import renderer from 'react-test-renderer';

const assertReactElement = element => {
  const component = renderer.create(element);
  return component.toJSON();
};

const colorOptions = [
  { value: 'red', label: 'Red' },
  { value: 'yellow', label: 'Yellow' },
  { value: 'green', label: 'Green' },
];

describe('CheckboxFieldComponent Component base', () => {
  it('should return a CheckboxFieldComponent that contains CheckboxComponent object', () => {
    const selectedValue = ['red'];
    const element = <CheckboxFieldComponent id="exampleColor" name="Color" options={colorOptions} value={selectedValue} />;
    expect(assertReactElement(element)).toMatchSnapshot();
  });
});
