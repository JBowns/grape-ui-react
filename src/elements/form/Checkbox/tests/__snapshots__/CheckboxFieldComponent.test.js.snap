// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxFieldComponent Component base should return a CheckboxFieldComponent that contains CheckboxComponent object 1`] = `
.c1 {
  color: inherit;
  background-color: transparent;
  font-family: inherit;
  font-size: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin-right: 1rem;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  cursor: pointer;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 1rem 0.5rem;
}

<div
  className="c0"
  id=""
>
  <label
    className="c1"
    id="exampleColor"
    value={
      Array [
        "red",
      ]
    }
  >
    <input
      checked={true}
      disabled={undefined}
      id="exampleColor"
      name="Color"
      onChange={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "marginRight": "0.5rem",
        }
      }
      type="checkbox"
      value="red"
    />
    Red
  </label>
  <label
    className="c1"
    id="exampleColor"
    value={
      Array [
        "red",
      ]
    }
  >
    <input
      checked={false}
      disabled={undefined}
      id="exampleColor"
      name="Color"
      onChange={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "marginRight": "0.5rem",
        }
      }
      type="checkbox"
      value="yellow"
    />
    Yellow
  </label>
  <label
    className="c1"
    id="exampleColor"
    value={
      Array [
        "red",
      ]
    }
  >
    <input
      checked={false}
      disabled={undefined}
      id="exampleColor"
      name="Color"
      onChange={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "marginRight": "0.5rem",
        }
      }
      type="checkbox"
      value="green"
    />
    Green
  </label>
</div>
`;
